ui/src/components/ServiceMapModal/ServiceMapModal.tsx

```diff
@@ -0,0 +1,282 @@
+        color: '#000',
+    };
+
+    const getServiceMapData = useCallback(async () => {
+        try {
+            setIsLoading(true)
+
+            const serviceMapData: ServiceMapGraph = await api.serviceMapData()
+            
+            if (serviceMapData.nodes != null) {
+                for (let i = 0; i < serviceMapData.nodes.length; i++) {
+                    graphData.nodes.push({
+                        id: serviceMapData.nodes[i].id,
+                        value: serviceMapData.nodes[i].count,
+                        label: serviceMapData.nodes[i].name,
+                        title: "Count: " + serviceMapData.nodes[i].name,
+                    });
+                }
+            }
+
+            if (serviceMapData.edges != null) {
+                for (let i = 0; i < serviceMapData.edges.length; i++) {
+                    graphData.edges.push({
+                        from: serviceMapData.edges[i].source.id,
+                        to: serviceMapData.edges[i].destination.id,
+                        value: serviceMapData.edges[i].count,
+                        label: serviceMapData.edges[i].count.toString(),
+                        color: getProtocolColor(serviceMapData.edges[i].destination.protocol),
+                    });
+                }
+            }
+
+            setGraphData(graphData)
+            setIsLoading(false)
+
+        } catch (error) {
+            setIsLoading(false)
+            console.error(error);
+        }
+    }, [graphData])
+
+    useEffect(() => {
+        getServiceMapData()
+    }, [getServiceMapData])
+
+    const resetServiceMap = debounce(async () => {
+        try {
+            const serviceMapResetResponse = await api.serviceMapReset();
+            if (serviceMapResetResponse["status"] === "enabled") {
+                refreshServiceMap()
+            }
+
+        } catch (error) {
+            console.error(error);
+        }
+    }, 500);
+
+    const refreshServiceMap = debounce(() => {
+        // close and re-open modal
+        onClose()
+        onOpen()
+    }, 500);
+
+
+    return (
+        <Modal
+            aria-labelledby="transition-modal-title"
+            aria-describedby="transition-modal-description"
+            open={isOpen}
+            onClose={onClose}
+            closeAfterTransition
+            BackdropComponent={Backdrop}
+            BackdropProps={{
+                timeout: 500,
+            }}
+            style={{ overflow: 'auto' }}
+        >
+            <Fade in={isOpen}>
+                <Box sx={modalStyle}>
+                    {isLoading && <div className={spinnerStyle.spinnerContainer}>
+                        <img alt="spinner" src={spinnerImg} style={{ height: 50 }} />
+                    </div>}
+                    {!isLoading && <div>
+                        <Button
+                            variant="contained"
+                            style={{
+                                margin: "0px 0px 0px 0px",
```

Comment:
all the buttons have the same style - make a common style object for all of them or use the existing style object we have for button in the app